gui/create-item.lua

```diff
@@ -448,12 +450,12 @@ function hackWish(unit)
     if df.item_type.attrs[itemtype].is_stackable then
      createItem({mattype,matindex},{itemtype,itemsubtype},quality,unit,description,amount)
     else
-     local isCorpsePiece = itemtype == df.item_type.CORPSEPIECE
+     local isCorpsePiece = itemtype ~= df.item_type.CORPSEPIECE or itemtype ~= df.item_type.CORPSE
      for i=1,amount do
       if not isCorpsePiece then
        createItem({mattype,matindex},{itemtype,itemsubtype},quality,unit,description,1)
       else
-       createCorpsePiece(unit,bodypart-2,partlayerID-1,mattype,matindex,corpsepieceGeneric,quality)
+       createCorpsePiece(unit,bodypart-2,partlayerID-1,mattype,matindex,corpsepieceGeneric,quality) --the offsets here are cause indexes in lua are fucky (some start at 0, some start at 1), so we adjust for that, as well as the index offset created by inserting the "generic" option at the start of the body part selection prompt
```

Comment:
No cursing in code, please.

Also, it sounds like the subtraction is a mistake if the index is invalid (i.e. equal to -1)

Reply:
whoops.

also -1 is what i use to let the function know that it's a whole corpse (i.e. there's no specific material/tissue to spawn so just spawn it all)

Reply:
does sound a bit hacky tho, but i'm trying to avoid adding too many variables that only get used in specific cases.

Reply:
just got what you meant, whoops.